scripts/docs_update_release.py

```diff
@@ -0,0 +1,216 @@
+        else:   # add new installer button to the top of the offline installer buttons
+            first_occurrence = selected_elements[0][0]
+
+            print(f"First occurrence on line {first_occurrence} - adding new installer button here")
+            index_data = index_lines[0:first_occurrence-1] + list(_resolve_installer_type()+'\n') + index_lines[first_occurrence:]
+            new_index_data = ''.join(index_data)
+    else:   # replace the first occurrence of the other installer type button
+        new_index_data = _replace_installer_button(selected_elements[0])
+
+
+    try:
+        with open(path.abspath(INDEX_PATH), "w") as index_file:
+            index_file.write(new_index_data)
+    except FileNotFoundError as e:
+        raise SystemExit(f"Error writing file {INDEX_PATH} - {e}")
+
+
+
+def update_releases_json():
+    """Update the releases.json file with the new release of the installer"""
+    try:
+        with open(path.abspath(RELEASES_JSON_PATH), "r") as releases_file:
+            releases_data = releases_file.read()
+    except FileNotFoundError as e:
+        raise SystemExit(f"Error opening file {RELEASES_JSON_PATH} - {e}")
+
+    try:
+        releases_json = json.loads(releases_data)
+    except json.JSONDecodeError as e:
+        raise SystemExit(f"Error parsing json: {e}")
+
+    releases_json.insert(0, {
+            "version": str(new_idf_version) if installer_type != 'online' else str(online_installer_version),
+            "type": str(installer_type),
+            "date": str(date.today()),
+            "size": str(installer_size)
+        })
+
+    try:
+        with open(path.abspath(RELEASES_JSON_PATH), "w") as releases_file:
+            json.dump(releases_json, releases_file, indent=4)
+    except FileNotFoundError as e:
+        raise SystemExit(f"Error writing file {RELEASES_JSON_PATH} - {e}")
+
+
+def update_inno_setup():
+    """Update the version of the installer in the InnoSetup file"""
+    try:
+        with open(path.abspath(INNO_SETUP_PATH), "r") as inno_setup_file:
+            inno_setup_data = inno_setup_file.read()
+    except FileNotFoundError as e:
+        raise SystemExit(f"Error opening file {INNO_SETUP_PATH} - {e}")
+
+    if installer_type == 'online':
+        replaced = re.sub(r'#define MyAppVersion "(\d+\.\d+)"', f'#define MyAppVersion "{online_installer_version}"', inno_setup_data)
+    elif installer_type == 'espressif-ide':
+        replaced = re.sub(r'#define ESPRESSIFIDEVERSION "(\d+\.\d+\.\d+)"', f'#define ESPRESSIFIDEVERSION "{ide_version}"', inno_setup_data)
+
+    try:
+        with open(path.abspath(INNO_SETUP_PATH), "w") as inno_setup_file:
+            inno_setup_file.write(replaced)
+    except FileNotFoundError as e:
+        raise SystemExit(f"Error writing file {INNO_SETUP_PATH} - {e}")
+
+
+
+def main():
+    """Performs the update of all necessary files for the new release of the installer"""    
```

Comment:
Unfortunately, this is kinda "hack" because of GitHub Actions' reusable workflows.
For some reason, I haven't been able to pass the output variable into the caller workflow from the called workflow (where the size of the installer is calculated). So it is passed such as an artifact txt file.

And because this script is not supposed to be run other way than from the caller workflow I think it is not necessary to use a more complex argument parser than approach `arg` position directly.

Also if the argument or environmental variables are missing (or not in format) there is a print to standard output (and workflow fail) for a developer to correct this.